psse/psse-converter/src/main/java/com/powsybl/psse/converter/PsseImporter.java

```diff
@@ -291,38 +405,243 @@ private static void createLine(PsseNonTransformerBranch psseLine, ContainersMapp
+            double x23 = psseTfo.getSecondRecord().getX23();
+            double r31 = psseTfo.getSecondRecord().getR31();
+            double x31 = psseTfo.getSecondRecord().getX31();
+            if (psseTfo.getFirstRecord().getCz() == 2) {
+                //CZ = 2 change to right Sbase pu
+                r12 = r12 * sbase / sbase12;
+                x12 = x12 * sbase / sbase12;
+                r23 = r23 * sbase / sbase23;
+                x23 = x23 * sbase / sbase23;
+                r31 = r31 * sbase / sbase31;
+                x31 = x31 * sbase / sbase31;
+            } else if (psseTfo.getFirstRecord().getCz() == 3) {
+                //CZ = 3 convert load loss power and current into pu impedances
+                r23 = r23 * sbase / (sbase23 * sbase23 * 1000000);
+                r31 = r31 * sbase / (sbase23 * sbase23 * 1000000);
+
+                double absZ23 = x23 * sbase / sbase23;
+                if (absZ23 * absZ23 - r23 * r23 <= 0) {
+                    x23 = 0;
+                    LOGGER.warn("inductance x23 of Transformer ({}) set to 0 because impedance module is ({}) and resistance is ({})  ", id, absZ23, r23);
+                } else {
+                    x23 = Math.sqrt(absZ23 * absZ23 - r23 * r23);
+                }
+
+                double absZ31 = x31 * sbase / sbase31;
+                if (absZ23 * absZ23 - r23 * r23 <= 0) {
+                    x31 = 0;
+                    LOGGER.warn("inductance x31 of Transformer ({}) set to 0 because impedance module is ({}) and resistance is ({})  ", id, absZ31, r31);
+                } else {
+                    x31 = Math.sqrt(absZ31 * absZ31 - r31 * r31);
+                }
+            }
+
+            //transform triangle (rij,xij) impedances into star (ri,xj) impedances
+            double sumR = r12 + r23 + r31;
+            double sumX = x12 + x23 + x31;
+            double squareMod = sumR * sumR + sumX * sumX;
+
+            double r1 = ((r31 * r12 - x31 * x12) * sumR + (r31 * x12 + r12 * x31) * sumX) / squareMod;
+            double x1 = ((r31 * x12 + r12 * x31) * sumR - (r31 * r12 - x31 * x12) * sumX) / squareMod;
+
+            double r2 = ((r12 * r23 - x12 * x23) * sumR + (r12 * x23 + r23 * x12) * sumX) / squareMod;
+            double x2 = ((r12 * x23 + r23 * x12) * sumR - (r12 * r23 - x12 * x23) * sumX) / squareMod;
+
+            double r3 = ((r23 * r31 - x23 * x31) * sumR + (r23 * x31 + r31 * x23) * sumX) / squareMod;
+            double x3 = ((r23 * x31 + r31 * x23) * sumR - (r23 * r31 - x23 * x31) * sumX) / squareMod;
+
+            //set a voltage base at star node with the associated Zbase
+            double v0 = 1.0;
+            double zbV0 = v0 * v0 / perUnitContext.getSb();
+
+            ThreeWindingsTransformerAdder tfoAdder = voltageLevel1.getSubstation().newThreeWindingsTransformer()
+                    .setRatedU0(v0)
+                    .setId(id);
+
+            ThreeWindingsTransformerAdder.LegAdder l1adder = tfoAdder.newLeg1();
```

Comment:
Same remark than with shunts, it could be written without adder vars in a fluent way.